src/qoi/src/spectroscopic_absorption.C

```diff
@@ -63,4 +63,12 @@ namespace GRINS
     QoIBase::_qoi_value = sys_qoi;
   }
 
+  void SpectroscopicAbsorption::finalize_derivative(libMesh::NumericVector<libMesh::Number> & derivatives)
+  {
+    if (!derivatives.closed())
+      derivatives.close();
+
+    derivatives.scale(-100.0 * QoIBase::_qoi_value);
```

Comment:
Found a problem here: apparently the adjoint residual error estimator doesn't compute the QoI value before doing the error estimate, meaning that `QoIBase::_qoi_value == 0` and we converge without doing any AMR.

I can reorder the call sequence in `MeshAdaptiveSolver` to calculate the QoI value before doing the error estimate, but it feels a bit hacky to me.

I'm unsure of the proper way to modify the AMR code to account for when the QoI error estimate depends explicitly on the QoI value.

Reply:
Capturing discussion from meat space. I told @tradowsk he should cache a reference to the `MultiphysicsSystem` and then called `assemble_qoi` here to populate the QoI value. We'll see later how big of penalty this is and, if need be, we can add hooks to prevent computing the QoI value more than once, but I don't think we'll need to.

Reply:
Looks like the easiest way to do this is through `IntegratedFunction::init()` since we get a reference to the `MultiphysicsSystem` as a parameter. I can't do it through the `SpectroscopicAbsorption` constructor since `QoIFactory` does not have access to the system. So `IntegratedFunction` would then be the class to cache the `MultiphysicsSystem` reference